core/include/drivers/regulator.h

```diff
@@ -76,13 +80,19 @@ struct regulator {
  * @get_state: Get regulator effective state
  * @set_voltage: Set voltage level in microvolt (uV)
  * @get_voltage: Get current voltage in microvolt (uV)
+ * @supported_voltages: Get an array of supported levels in microvolt (uV)
+ *      When successful, @count is the number of cells in @levels_us
+ *      or @count is 0 when @levels_us is a 3 cells array [min, max, step],
+ *      in microvolt, for linear incremental level lists.
```

Comment:
I am not very comfortable with `count == 0` for a 3-cell array, and the fact that the array is not constant format nor self-describing. Wouldn't it be a bit easier to define a struct?
```C
/* Defines the format of struct voltages::entries */
enum voltage_type {
        VOLTAGE_TYPE_FULL_LIST, /* extensive list in uV */
        VOLTAGE_TYPE_INCREMENT  /* min, max, increment (in uV) */
};
struct voltages {
        enum voltage_type type;
        size_t num_levels;
        /* Dynamic part: num_levels entries */
        /* int entries[] */
};
```

Reply:
ok, makes sense. I'll update.
Thanks for the proposal.